web/src/api/images/verify_app.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import { protectInternalEndpoint } from "src/backend/utils";
+import { getAPIReviewerGraphqlClient } from "src/backend/graphql";
+import { getSdk as getAppMetadataSDK } from "@/api/images/graphql/getAppMetadata.generated";
+import { getSdk as deleteVerifiedAppMetadataSDK } from "@/api/images/graphql/deleteVerifiedAppMetadata.generated";
+import { getSdk as updateAppMetadataSDK } from "@/api/images/graphql/updateAppMetadata.generated";
+import { getFileExtension } from "src/backend/utils";
+
+import {
+  S3Client,
+  ListObjectsCommand,
+  PutObjectTaggingCommand,
+  CopyObjectCommand,
+} from "@aws-sdk/client-s3";
+import { logger } from "@/lib/logger";
+import { randomUUID } from "crypto";
+
+export type VerifyAppResponse = {
+  success?: boolean;
+};
+
+/**
+ * This function handles the verification of the app by a reviewer
+ * @param req Expects an app_id and reviewer_name is_reviewer_app_store_approved and is_reviewer_world_app_approved
+ * @param res Returns a success message if the app was verified
+ */
+export const handleVerifyApp = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  try {
+    if (!protectInternalEndpoint(req, res)) {
+      return;
+    }
+
+    if (req.method !== "POST") {
+      return errorNotAllowed(req.method!, res, req);
+    }
+
+    if (
+      !["reviewer", "admin"].includes(
+        req.body.session_variables["x-hasura-role"]
+      )
+    ) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "Unauthorized Role",
+        code: "unauthorized_role",
```

Comment:
Let's log the error but we should not send details back for security reason, let's remove the `detail` & `code` to send a generic error.